source/modulo_components/modulo_components/component_interface.py

```diff
@@ -67,29 +67,159 @@ def __step(self):
             self._predicate_publishers[predicate_name].publish(msg)
 
     def __generate_predicate_topic(self, predicate_name: str) -> str:
-        return f'/predicates/{self.get_name()}/{predicate_name}'
+        """
+        Generate the predicate topic name from the name of the predicate.
+
+        :param predicate_name: The predicate name
+        :return: The predicate topic as /predicates/component_name/predicate_name
+        """
+        return f"/predicates/{self.get_name()}/{predicate_name}"
+
+    def add_parameter(self, parameter: Union[str, sr.Parameter], description: str, read_only=False) -> None:
+        """
+        Add a parameter. This method stores either the name of the attribute corresponding to the parameter object or
+        a parameter object directly in the local parameter dictionary and declares the equivalent ROS parameter on the
+        ROS interface. If an attribute name is provided, changing the ROS parameter will also update the provided
+        attribute and vice versa. If the provided parameter is not an attribute name or a Parameter object, does not
+        do anything.
+
+        :param parameter: Either the name of the parameter attribute or the parameter itself
+        :param description: The parameter description
+        :param read_only: If True, the value of the parameter cannot be changed after declaration
+        """
+        try:
+            if isinstance(parameter, sr.Parameter):
+                sr_parameter = parameter
+            elif isinstance(parameter, str):
+                sr_parameter = self.__getattribute__(parameter)
```

Comment:
It seems we should also check for the type of the attribute, otherwise `write_parameter` or `sr_parameter.get_name()` could fail. 

```suggestion
                attr = self.__getattribute__(parameter)
                if isinstance(attr, sr.Parameter):
                    sr_parameter = attr
                else:
                    raise ...
```